Tests/MapboxMapsTests/Snapshot/SnapshotterTests.swift

```diff
@@ -1,28 +1,166 @@
 import XCTest
 @testable import MapboxMaps
+@_implementationOnly import MapboxCoreMaps_Private
+@_implementationOnly import MapboxCommon_Private
+import CoreLocation
 
 final class SnapshotterTests: XCTestCase {
 
     var mapboxObservableProviderStub: Stub<ObservableProtocol, MapboxObservableProtocol>!
     var snapshotter: Snapshotter!
+    var mockMapSnapshotter: MockMapSnapshotter!
 
     override func setUp() {
         super.setUp()
         let options = MapSnapshotOptions(
             size: CGSize(width: 100, height: 100),
             pixelRatio: .random(in: 1...3))
         mapboxObservableProviderStub = Stub(defaultReturnValue: MockMapboxObservable())
+        mockMapSnapshotter = MockMapSnapshotter()
         snapshotter = Snapshotter(
             options: options,
-            mapboxObservableProvider: mapboxObservableProviderStub.call(with:))
+            mapboxObservableProvider: mapboxObservableProviderStub.call(with:),
+            mapSnapshotter: mockMapSnapshotter)
     }
 
     override func tearDown() {
         snapshotter = nil
         mapboxObservableProviderStub = nil
+        mockMapSnapshotter = nil
         super.tearDown()
     }
 
+    // Test snapshot start invokes mockMapSnapshotter startStub
+    func testSnapshotterStartInvocation() {
+        snapshotter.start(overlayHandler: nil) { (_) in
+            XCTAssertNotNil(self.mockMapSnapshotter.startStub.defaultReturnValue)
+        }
+
+        XCTAssertEqual(mockMapSnapshotter.startStub.invocations.count, 1)
+    }
+
+    func testSnapshotterCompletionInvocationFailed() {
+
+        let options = MapSnapshotOptions(size: CGSize.init(width: 300, height: 300), pixelRatio: 2)
+        snapshotter = Snapshotter(options: options, mapboxObservableProvider: mapboxObservableProviderStub.call(with:), mapSnapshotter: mockMapSnapshotter)
```

Comment:
do we need this? snapshotter is initialized before every test cases, and I don't see anything special for this test case to have a different snapshoter.